File: lc_soln/0169_majority_element/solution.cpp
#include <vector>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int count = 1;
        int last = nums[0];
        for (int i = 1; i < nums.size(); ++i) {
            int num = nums[i];
            if (num == last) {
                ++count;
            } else {
                --count;
                if (count == 0) {
                    last = num;
                    count = 1;
                }
            }
        }
        return last;
    }
};

File: lc_soln/0169_majority_element/solution.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 1
        last = nums[0]
        for i in range(1, len(nums)):
            num = nums[i]
            if num == last:
                count += 1
            else:
                count -= 1
                if count == 0:
                    last = num
                    count = 1
        return last
